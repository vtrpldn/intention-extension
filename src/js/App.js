import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import Form from './components/Form/Form'
import styled from 'styled-components'
import GlobalStyle from './GlobalStyle';

const Wrapper = styled.div`
  display: ${({ display }) => display};
  background: rgba(255,255,255, 0.9);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 2147483647; /* NONE SHALL PASS */
  transition: all 1s ease;
`

class App extends Component {
  constructor() {
    super()
    this.state = {
      style: {
        display: 'block'
      }
    }
  }

  hideOverlay() {
    this.setState({
      style: {
        display: 'none'
      }
    })
  }

  render() {
    const { display } = this.state.style

    return (
      <div>
        <GlobalStyle />
        <Wrapper display={display}>
          <Form hideOverlay={() => this.hideOverlay()} />
        </Wrapper>
      </div>
    )
  }
}

chrome.runtime.sendMessage({ type: 'GET_PAGE_STATUS' }, function ({
  isCurrentUrlActive,
  isCurrentUrlOnList
}) {
  console.log('isCurrentUrlActive:', isCurrentUrlActive)
  console.log('isCurrentUrlOnList:', isCurrentUrlOnList)
  if (isCurrentUrlOnList === true && isCurrentUrlActive === false) {
    const app = document.createElement('div')
    document.body.appendChild(app)
    ReactDOM.render(<App />, app)
  }
})
