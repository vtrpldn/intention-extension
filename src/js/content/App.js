import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import Form from './components/Form/Form'
import styled from 'styled-components'

class App extends Component {
  constructor () {
    super()
    this.state = {
      display: 'block'
    }
  }

  hideOverlay() {
    this.setState({
      display: 'none'
    })
  }

  render () {
    const Wrapper = styled.div`
      display: ${this.state.display};
      background: white;
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      z-index: 2147483647;
    `

    return (
      <Wrapper>
        <Form hideOverlay={this.hideOverlay.bind(this)}/>
      </Wrapper>
    )
  }
}

chrome.runtime.sendMessage({ type: 'GET_PAGE_STATUS' }, function ({
  isCurrentUrlActive,
  isCurrentUrlOnList
}) {
  console.log('isCurrentUrlActive:', isCurrentUrlActive)
  console.log('isCurrentUrlOnList:', isCurrentUrlOnList)
  if (isCurrentUrlOnList === true && isCurrentUrlActive === false) {
    const app = document.createElement('div')
    document.body.appendChild(app)
    ReactDOM.render(<App />, app)
  }
})
