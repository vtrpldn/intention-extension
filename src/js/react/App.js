import React from 'react'
import ReactDOM from 'react-dom'

const App = () => {
  return (
    'Hi, my name is app'
  )
}

const app = document.createElement('div')
document.body.appendChild(app)
ReactDOM.render(<App />, app)

export default App
