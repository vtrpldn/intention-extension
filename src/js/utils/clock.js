export const secondToClock = (seconds) => {
  let date = new Date(null)
  date.setSeconds(seconds)
  const timeString = date.toISOString().substr(11, 8)
  return timeString
}