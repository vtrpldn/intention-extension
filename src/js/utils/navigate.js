export const navigationTarget = {
  NEW_WINDOW: 'new-window',
  NEW_TAB: 'new-tab',
  CURRENT_TAB: 'current-tab'
}

export const navigate = (url, target = navigationTarget.NEW_TAB) => {
  switch (target) {
    case navigationTarget.NEW_WINDOW:
      return chrome.windows.create({ url, focused: true, type: 'normal' })
    case navigationTarget.CURRENT_TAB:
      return chrome.tabs.update({ url, active: true })
    default:
      return chrome.tabs.create({ url, active: true })
  }
}
