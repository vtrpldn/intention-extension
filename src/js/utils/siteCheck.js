export const getUrlListStatus = (list, url) => {
  console.log('list:', list)
  console.log('url:', url)
  if (list === []) {
    return false
  }
  return !!list.filter((v) => url.includes(v)).length
}

export const getUrlActiveStatus = (activeSites, url) => {
  console.log('activeSites:', activeSites)
  console.log('url:', url)
  if (activeSites === []) {
    return false
  }
  return !!activeSites.map(v => v.url).filter((v) => v === url).length
}
