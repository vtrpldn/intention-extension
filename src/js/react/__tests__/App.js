import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'
import '../../utils/enzyme'

describe('App component testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
