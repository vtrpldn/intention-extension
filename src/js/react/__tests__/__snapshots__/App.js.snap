// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
